CONTRIBUTING.md

```diff
@@ -0,0 +1,69 @@
+# Contributing guide
+
+**Want to contribute? Great!** 
+We try to make it easy, and all contributions, even the smaller ones, are more than welcome.
+This includes bug reports, fixes, documentation, examples... 
+But first, read this page.
+
+## Legal
+
+All original contributions to Narayana are licensed under the
+[GNU LESSER GENERAL PUBLIC LICENSE Version 2.1](https://github.com/jbosstm/narayana/blob/master/LICENSE).
+
+All contributions are subject to the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).
+The DCO text is also included verbatim in the [dco.txt](dco.txt) file in the root directory of the repository.
+
+## Reporting an issue
+
+This project uses the [JBTM issue tracker](https://issues.redhat.com/projects/JBTM)
+to manage the issues. Open an issue directly in this issue tracker.
+
+If you believe you found a bug, and it's likely possible, please indicate a way to reproduce it, what you are seeing and what you would expect to see.
+Don't forget to indicate your Narayana, Java and Maven version. 
+
+### Building master
+
+See the [README](README.md) file in the root directory of the repository.
+
+## Before you contribute
+
+To contribute, use GitHub Pull Requests, from your **own** fork.
+
+Also, make sure you have set up your Git authorship correctly:
+
+```
+git config --global user.name "Your Full Name"
+git config --global user.email your.email@example.com
+```
+
+If you use different computers to contribute, please make sure the name is the same on all your computers.
+
+### Code reviews
+
+All submissions, including submissions by project members, need to be reviewed before being merged.
+
+### Guidelines
+
+We follow [Open source participation guidelines](https://www.redhat.com/en/resources/open-source-participation-guidelines-overview)
+
+### Coding Guidelines
+
+We use a checkstyle plugin which is skipped by default because we have a lot of old files, that we don't want to change, which would fail the compilation.
+But on new code we do encourage/advise that it should follow our checkstyle rules.
+
+To see if your code follows our style guide you have two options:
+
+Enable the checkstyle plugin via the pom file, or
```

Comment:
I think this could be hard to do for a developer (but I don't know how to better write the sentence, just saying)
My context is: 
* the checkstyle plugin is configured in root pom.xml
* it could be switched on by setting property on compilation `-Dcheckstyle.skip=false`
* when set up during compilation then it just completely fails as most of the code will not comply, ie. it's difficult to create a new file and try to check with checkstyle if it's ok
* setting up the property probably makes sense only in IDE
* the new modules - only the LRA does so now - enable the checkstyle and the checkstyle is verified during compilation automatically

Reply:
Great suggestion.